### PR TITLE
Resolve PHPStan Level 0 errors

### PR DESCRIPTION
**Pull request by szepeviktor**
@JJJ Please see comments.

### CODE REVIEW

File: src/Database/Query.php [hunk 1]
```diff
@@ -1524,7 +1524,7 @@ private function parse_where_join( $args = array() ) {
 		// Maybe perform a date query
 		if ( ! empty( $date_query ) && is_array( $date_query ) ) {
 			$this->date_query = $this->get_date_query( $date_query );
-			$clauses          = $this->date_query->get_sql( $this->table_name, $alias, $primary, $this );
+			$clauses          = $this->date_query->get_sql();
```
Comment (szepeviktor) [inline src/Database/Query.php:1527]: `Queries\Date::get_sql()` needs no parameters.

File: src/Database/Schema.php [hunk 2]
```diff
@@ -51,7 +51,7 @@ class Schema extends Base {
 	 * @since 1.0.0
 	 * @var   array
 	 */
-	protected $columns = array();
+	public $columns = array();
```
Comment (szepeviktor) [inline src/Database/Schema.php:54]:
```
src/Database/Schema.php:94:     if ( ! empty( $this->columns ) && is_array( $this->columns ) ) {
src/Database/Schema.php:95:         $this->setup_items( 'columns', $this->column, $this->columns );
src/Database/Schema.php:146:            $this->columns = array();
```

File: src/Database/Query.php (hunk 1 above)
Comment (JJJ) [reply src/Database/Query.php:1527]: You are correct.

It is like this as part of the requirements of #55.

Is it possible to have PHPStan ignore this, for now?

File: src/Database/Schema.php (hunk 2 above)
Comment (JJJ) [reply src/Database/Schema.php:54]: Hmm. Yeah. Same for `$indexes` too probably. Thanks!

File: src/Database/Query.php (hunk 1 above)
Comment (szepeviktor) [reply src/Database/Query.php:1527]: > You are correct.

@phpstan is correct!

**Comment by szepeviktor** [comment]
Done. Both. 🍏

**Comment by JJJ** [comment]
@szepeviktor any suggestions on using with Php 8.1+?

**Comment by szepeviktor** [comment]
> any suggestions on using with Php 8.1+

Do you mean running PHPStan on PHP 8.1 or running BerlinDB on PHP 8.1?

**Comment by JJJ** [comment]
> Do you mean running PHPStan on PHP 8.1 or running BerlinDB on PHP 8.1?

Both! 🫠 Running PHPStan (like you would normally) but for BerlinDB with PHP 8.1 as the active CLI version?

```php
Deprecated: Return type of PHPStan\Dependency\ExportedNode\ExportedPropertyNode::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/jjj/Development/JH/wp-content/plugins/berlindb/vendor/phpstan/phpstan/phpstan.phar/src/Dependency/ExportedNode/ExportedPropertyNode.php on line 71
Deprecated: Return type of PHPStan\Dependency\ExportedNode\ExportedInterfaceNode::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/jjj/Development/JH/wp-content/plugins/berlindb/vendor/phpstan/phpstan/phpstan.phar/src/Dependency/ExportedNode/ExportedInterfaceNode.php on line 56
Deprecated: Return type of PHPStan\Dependency\ExportedNode\ExportedPhpDocNode::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/jjj/Development/JH/wp-content/plugins/berlindb/vendor/phpstan/phpstan/phpstan.phar/src/Dependency/ExportedNode/ExportedPhpDocNode.php on line 37
Deprecated: Return type of PHPStan\Dependency\ExportedNode\ExportedFunctionNode::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/jjj/Development/JH/wp-content/plugins/berlindb/vendor/phpstan/phpstan/phpstan.phar/src/Dependency/ExportedNode/ExportedFunctionNode.php on line 73
Deprecated: Return type of PHPStan\Dependency\ExportedNode\ExportedClassNode::jsonSerialize() should either be compatible with JsonSerializable::jsonSerialize(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/jjj/Development/JH/wp-content/plugins/berlindb/vendor/phpstan/phpstan/phpstan.phar/src/Dependency/ExportedNode/ExportedClassNode.php on line 85
```

**Comment by szepeviktor** [comment]
As `mixed` typehint is not supported before PHP 8.1 the only way to resolve this is adding attribute `#[\ReturnTypeWillChange]`

**Pull request merged**